test-integ/peering_commontopo/ac5_2_test.go

```diff
@@ -188,115 +155,111 @@ func (s *preparedQueryFailoverSuite) setupForDC3(ct *commonTopo, clu, peer1, pee
 	}
 
 	serverNode := ct.AddServiceNode(clu, server)
-
-	s.dc3ServerSID = serverSID
-	s.nodeServer = append(s.nodeServer, serverNode)
+	serverNode.Disabled = true
 }
 
-func (s *preparedQueryFailoverSuite) test(t *testing.T, ct *commonTopo) {
-	var partition = "default"
-	dc := ct.Sprawl.Topology().Clusters[s.DC]
-	apiClient := ct.APIClientForCluster(t, dc)
-
-	dc3 := ct.Sprawl.Topology().Clusters[ct.DC3.Name]
-	s.dc3APIClient = ct.APIClientForCluster(t, dc3)
+func (s *preparedQueryFailoverSuite) createPreparedQuery(t *testing.T, ct *commonTopo, c *api.Client, partition string) (*api.PreparedQueryDefinition, *api.PreparedQuery) {
+	// create prepared query definition
+	var (
+		peers []string
+		err   error
+	)
+	peers = append(peers, LocalPeerName(ct.DC2, partition), LocalPeerName(ct.DC3, partition))
+
+	def := &api.PreparedQueryDefinition{
+		Name: "ac5-prepared-query",
+		Service: api.ServiceQuery{
+			Service:     s.serverSID.Name,
+			Partition:   ConfigEntryPartition(s.serverSID.Partition),
+			OnlyPassing: true,
+			Failover: api.QueryFailoverOptions{
+				Targets: func() []api.QueryFailoverTarget {
+					var queryFailoverTargets []api.QueryFailoverTarget
+					for _, peer := range peers {
+						queryFailoverTargets = append(queryFailoverTargets, api.QueryFailoverTarget{
+							Peer: peer,
+						})
+					}
+					return queryFailoverTargets
+				}(),
+			},
+		},
+	}
 
-	// get peer names for dc2 & dc3 cluster respectively
-	var peers []string
-	peers = append(peers, LocalPeerName(ct.DC2, partition))
-	peers = append(peers, LocalPeerName(ct.DC3, partition))
+	query := c.PreparedQuery()
+	def.ID, _, err = query.Create(def, nil)
+	require.NoError(t, err, "error creating prepared query in %s")
 
-	s.testServiceHealthCheck(t, apiClient)
-	s.testQueryTwoFailovers(t, apiClient, ct, peers)
+	return def, query
 }
 
-func (s *preparedQueryFailoverSuite) testServiceHealthCheck(t *testing.T, apiClient *api.Client) {
-	t.Run("validate service health in cluster", func(t *testing.T) {
-		// preconditions check
-		assertServiceHealth(t, s.dc3APIClient, s.serverSID.Name, 1) // validate server is healthy in dc3 cluster
-		assertServiceHealth(t, apiClient, s.serverSID.Name, 1)
-	})
-}
-
-func (s *preparedQueryFailoverSuite) testQueryTwoFailovers(t *testing.T, c *api.Client, ct *commonTopo, peers []string) {
-	if s.DC == "dc2" {
-		// TO-DO: currently failing
-		t.Skip()
+func (s *preparedQueryFailoverSuite) test(t *testing.T, ct *commonTopo) {
+	dc1 := ct.Sprawl.Topology().Clusters["dc1"]
+	dc2 := ct.Sprawl.Topology().Clusters["dc2"]
+	dc3 := ct.Sprawl.Topology().Clusters["dc3"]
+	dc3Client := ct.APIClientForCluster(t, dc3)
+
+	def, query := s.createPreparedQuery(t, ct, dc3Client, "default")
+
+	type testcase struct {
+		name      string
+		cluster   *topology.Cluster
+		peer      *topology.Cluster
+		partition string
 	}
+	tcs := []testcase{
+		{
+			name:      "validate service health in cluster",
```

Comment:
Other than the `name`, this case is the same as item 2?

Also, I'm not sure the `name`s are accurate? Looks it always validates service health, and always does `testQueryTwoFailovers`.